Gintoki

g1nt0ki opened the pull request.

princeraj-nord commented:
@g1nt0ki We had implemented a fetch adapter and you can refer to the below PR for the same.
https://github.com/DefiLlama/DefiLlama-Adapters/pull/2221

We can see that the complete code of Nord finance adapter changed in this PR and merged without our review. These changes include SDK adapter type which doesn't fits in our case that's the reason we have implemented the fetch adapter. Please refer to this PR also for our previous conversation about SDK and fetch adapter.
https://github.com/DefiLlama/DefiLlama-Adapters/pull/1836

Please let me know how we can fix this now.

g1nt0ki commented:
> @g1nt0ki @waynebruce0x We had implemented a fetch adapter for Nord Finance and you can refer to the below PR for the same. #2221
> 
> We can see that the complete code of Nord finance adapter changed in this PR and merged without our review. These changes include SDK adapter type which doesn't fits in our case that's the reason we have implemented the fetch adapter. Please refer to this PR also for our previous conversation about SDK and fetch adapter. #1836
> 
> Please let me know how we can fix this now.

Hi princeraj, 

Sorry, I am not a big fan of fetch adapter, think I had to re-write it few days ago because the call to the endpoint was being blocked by cloudflare/firewall on your side. Can you tell me what you dislike about my implementation, am I am missing some tokens?

We do not like fetch adapters in general because it is opaque and might fail sometime in the future and we have to run after protocol devs

jvnord commented:
@g1nt0ki Our savings protocol is yield aggregator. If we go by SDK adapter, it will never show the correct TVL for our protocol as we invest user funds into different yield-bearing protocols such as aave, compound on different chains. Hence the fetch adapter is the only way we can show our TVL.

Let us know how we can move forward here.

g1nt0ki commented:
hi @princeraj-nord, I understand, is there a way to fetch token deposits from on chain calls? some of the yield aggregators support this.
